src/util/address.rs

```diff
@@ -467,32 +467,80 @@ mod tests {
 
 use crate::vm::VMBinding;
 
-/// ObjectReference represents address for an object. Compared with Address,
-/// operations allowed on ObjectReference are very limited. No address arithmetics
-/// are allowed for ObjectReference. The idea is from the paper
-/// High-level Low-level Programming (VEE09) and JikesRVM.
+/// `ObjectReference` represents address for an object. Compared with `Address`, operations allowed
+/// on `ObjectReference` are very limited. No address arithmetics are allowed for `ObjectReference`.
+/// The idea is from the paper [Demystifying Magic: High-level Low-level Programming (VEE09)][FBC09]
+/// and [JikesRVM].
+///
+/// In MMTk, `ObjectReference` holds a non-zero address.  It either points into one of MMTk's spaces
```

Comment:
This is a good place to specify the requirements of `ObjectReference`:
1. uniquely references an object for MMTk
2. within allocated memory (VM space is an exception which needs to be discussed separately)
3. easy to get object info from `ObjectReference`

Reply:
Probably don't say an `ObjectReference` 'points into one of MMTk's spaces'. When we have contiguous spaces, the entire address range is considered as the MMTk space, but not any address that is in the range can be an `ObjectReference`. An `ObjectReference` has to point inside the allocated memory.

Reply:
I added uniqueness to the doc, and mentioned that it must be easy to get object info from `ObjectReference`.  I also added a section for `VMSpace`.

I now say "`ObjectReference` that refers to an MMTk object" instead of "`ObjectReference" that points into one of MMTk's spaces".